src/components/TransactionStatus.tsx

```diff
@@ -101,29 +120,37 @@ function TransactionStatus({ transaction, onComplete }: Props) {
       return;
     }
 
-    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null) => {
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null): boolean => {
       if (!transactionValue || !chainValue) return false;
 
       const bnChainValue = new BN(chainValue);
       const bnTransactionValue = new BN(transactionValue);
       return bnChainValue.gt(bnTransactionValue);
     };
 
-    const completionStatus = (status: boolean) =>
-      transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+    const completionStatus = (status: boolean = false): TransactionStatusEnum => {
+      if (transaction.id === 0) {
+        return TransactionStatusEnum.NOT_STARTED;
+      }
+      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+    };
```

Comment:
```suggestion
    const completionStatus = (isCompleted: boolean): TransactionStatusEnum => {
      if (transaction.id === 0) {
        return TransactionStatusEnum.NOT_STARTED;
      }
      return isCompleted ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
    };
```

Having a default of `false` doesn't really make much sense, cause the status would never be `COMPLETED`. I think setting it to `true` is more sensible, because you are basically saying: If only there is a `block` then it's done.

However, it's even better if we don't define default and simply require the caller to always pass the status. This function has less dependencies then (good) and also is more straightforward what it does, if you were to write documentation and describe it in a sentence it would be:
"This functions returns completion status of some a transaction. If the `transaction.id` is not know it's considered not started, and later it's state is controlled by `isCompleted` parameter."